src/bin/ion/commands/beta/inspect.rs

```diff
@@ -184,106 +191,32 @@ fn inspect_file(input_file_name: &str,
     match ion_data {
         // Pattern match the byte array to verify it starts with an IVM
         [0xE0, 0x01, 0x00, 0xEA, ..] => {
-            let mut inspector = IonInspector::new(
-                ion_data,
-                Rc::clone(output),
-                bytes_to_skip,
-                limit_bytes,
-            );
-
-            write_header(&output)?;
+            write_header(output)?;
+            let mut inspector = IonInspector::new(ion_data, output, bytes_to_skip, limit_bytes);
             // This inspects all values at the top level, recursing as necessary.
             inspector.inspect_level()?;
         }
         _ => {
             // bail! constructs an `anyhow::Result` with the given context and returns.
-            bail!("Input file '{}' does not appear to be binary Ion.", input_file_name);
+            bail!(
+                "Input file '{}' does not appear to be binary Ion.",
+                input_file_name
+            );
         }
     };
     Ok(())
 }
 
-// The ion::Reader type allows you to specify an event handler to react to low-level events in the
-// stream being read. This type summarizes them; it doesn't write out their full hex encoding,
-// it just writes a comment describing the event in the text Ion column.
-struct SystemLevelEventSummarizer {
-    output: OutputRef,
-    text_buffer: String,
-}
-
-impl SystemLevelEventSummarizer {
-    pub fn new(output: OutputRef) -> SystemLevelEventSummarizer {
-        SystemLevelEventSummarizer {
-            output,
-            text_buffer: String::with_capacity(512),
-        }
-    }
-}
-
 const IVM_HEX: &str = "e0 01 00 ea";
 const IVM_TEXT: &str = "// Ion 1.0 Version Marker";
 // System events (IVM, symtabs) are always at the top level.
 const SYSTEM_EVENT_INDENTATION: &str = "";
-
-impl SystemEventHandler for SystemLevelEventSummarizer {
```

Comment:
The new `SystemReader` in `ion-rs` obviated the need for the `SystemEventHandler` trait.